src/test/scala/chiselTests/SourceLocatorSpec.scala

```diff
@@ -41,4 +70,36 @@ class SourceLocatorSpec extends ChiselFunSpec with Utils {
       escapedInfo.unescaped should equal(filename)
     }
   }
+
+  describe("(2) Module source locators") {
+    it("(2.a): modules extending RawModule should have a source locator") {
+      val chirrtl = emitCHIRRTL(new RawModuleChild)
+      chirrtl should include(s"module RawModuleChild : @[$thisFile 14:9]")
+    }
+    it("(2.b): modules extending Module should have a source locator") {
+      val chirrtl = emitCHIRRTL(new ModuleChild)
+      chirrtl should include(s"module ModuleChild : @[$thisFile 15:9]")
+    }
+    it("(2.c): modules extending other user modules should have a source locator") {
+      val chirrtl = emitCHIRRTL(new InheritanceModule)
+      chirrtl should include(s"module InheritanceModule : @[$thisFile 16:9]")
+    }
+    it("(2.d): modules extending BlackBox should have a source locator") {
+      val chirrtl = emitCHIRRTL(new WrapperTop(new BlackBoxChild))
+      chirrtl should include(s"extmodule BlackBoxChild : @[$thisFile 17:9]")
+    }
+    it("(2.e): modules extending ExtModule should have a source locator") {
+      val chirrtl = emitCHIRRTL(new WrapperTop(new ExtModuleChild))
+      chirrtl should include(s"extmodule ExtModuleChild : @[$thisFile 20:9]")
+    }
+    it("(2.f): user-defined Classes should have a source locator") {
+      val chirrtl = emitCHIRRTL(new ClassTop)
+      chirrtl should include(s"class ClassChild : @[$thisFile 24:9]")
+    }
+    it("(2.g): Inner and anonymous modules should have a source locators") {
+      val chirrtl = emitCHIRRTL(new Outer)
+      chirrtl should include(s"module Inner : @[$thisFile 29:11]")
+      chirrtl should include(s"module AnonymousModule : @[$thisFile 31:25]")
+    }
```

Comment:
awesome! 

are there any interactions with D/I or Instantiate that need to be tested?

Reply:
Definitions should get a source locator in the same way as these examples, I'll add a test!

Reply:
Added new test.